src/hyperfine/benchmark.rs

```diff
@@ -221,17 +231,21 @@ pub fn run_benchmark(
                 options.failure_action,
                 None,
             )?;
-            progress_bar.inc(1);
+            if let Some(pbar) = progress_bar.clone() { pbar.inc(1); }
         }
-        progress_bar.finish_and_clear();
+        if let Some(bar) = progress_bar { bar.finish_and_clear(); }
     }
 
     // Set up progress bar (and spinner for initial measurement)
-    let progress_bar = get_progress_bar(
-        options.runs.min,
-        "Initial time measurement",
-        options.output_style,
-    );
+    let mut progress_bar:Option<ProgressBar> = None;
+
+    if options.output_style != OutputStyleOption::None { 
+            progress_bar = Some(get_progress_bar(
+            options.runs.min,
+            "Initial time measurement",
+            options.output_style,
+        ))
+    }
```

Comment:
Same as above. can be constructed in a a single statement.

Reply:
updated